### PR TITLE
Add list item expansion with select based pie chart.

### PR DESCRIPTION
**Pull request by chambridge**
Add list item expansion with select based pie chart.

* Pulled portions of trend chart into a commonChart component to be reused with pieChart component
* Created pie chart component
* Created detailChart component to handle state and fetch report data
* Created detailItem component to build each expandable list item
* Updated costDetails to create detailItems

_Note: There is an issue with the coloring of the pie chart that I will continue to debug._

Expandable list item close state:
image

Expandable list item opened:
image

Change grouped data in expandable list item:
image

Multiple expanded list items of different groupings:
image

### CODE REVIEW

**Comment by dmiller9911** [comment]
Deploy preview for *koku* ready!

Built with commit ee665169c37c6afbcec65d17f4cd835a5279061e

https://deploy-preview-95--koku.netlify.com

**Comment by dlabrecq** [comment]
Deploy preview for *project-koku* ready!

Built with commit ee665169c37c6afbcec65d17f4cd835a5279061e

https://deploy-preview-95--project-koku.netlify.com

**Comment by chambridge** [comment]
I created the following bug to track the pie chart coloring issue:
https://github.com/project-koku/koku-ui/issues/96

File: src/components/pieChart/pieChart.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+  public render() {
+    const { title, height, width, data, groupBy } = this.props;
+
+    const currentData = transformReport(data, ChartType.monthly, groupBy);
+    const legendData = currentData.map(item => ({
+      name: item.name.toString() + ' (' + formatCurrency(item.y) + ')',
+      symbol: { type: 'square' },
+    }));
+    const colors = 'cool';
```
[the remark this answers is not in the document]
Comment (chambridge) [reply src/components/pieChart/pieChart.tsx]: I’m already applying colorScale. I don’t know if it’s a bug in the library or something else but it always just applies the first color to every slice. “cool” is a built in set that translates to an array of colors in the library. I have tried providing my own array to no effect. 
If you look at the legend it uses the same colorScale and should match up with the pie chart. 
Thanks for the suggestion, but unfortunately it doesn’t solve the problem.

[the remark this answers is not in the document]
Comment (chambridge) [reply src/components/pieChart/pieChart.tsx]: I’m already applying colorScale. I don’t know if it’s a bug in the library or something else but it always just applies the first color to every slice. “cool” is a built in set that translates to an array of colors in the library. I have tried providing my own array to no effect. 
If you look at the legend it uses the same colorScale and should match up with the pie chart. 
Thanks for the suggestion, but unfortunately it doesn’t solve the problem.

**Pull request merged**